app/pages/lab-pages-editor/helpers/linkStepsInWorkflow.js

```diff
@@ -0,0 +1,54 @@
+/*
+Links Steps in a Workflow together.
+
+- Input: array of Steps, and Tasks object.
+- Output: COPY of array of Steps, with their `step.next`s updated.
+
+Rules for linking steps:
+- If a Step *can* branch, then step.next = undefined.
+- If a Step is the final step in the array, then step.next = undefined. 
```

Comment:
My personal preference is to not conflate two purposes for a value of "undefined". An alternative is to make the final step = -1 or a somewhat nonsensical yet meaningful to the developer value that's easy to check. Not a requirement, but  I find it very useful to my future self figuring out how to debug something like this.